### ISSUE TITLE
Naughtyblog website down error

### ISSUE THREAD

**Issue by krtlsbrk** [dobbelina/repository.dobbelina#1455]
### Prerequisites

- [X] I have searched the issue tracker to avoid opening a duplicate issue.
- [X] I have read the [documentation](https://github.com/dobbelina/repository.dobbelina/wiki/Cumination:-Debug-Logging-and-Uploading) on how to create a debug log file.

### Description

Hello,

There seems to be an issue with naughtyblog in kodi. It throws an error saying that the website is down but the website is up.

### Kodi version

21

### Cumination version

1.0.4

### Operating System and version

Android TV Nvidia Shield Pro

**Comment by EvelK1203** [comment]
I think there is something going on with Kodi/Android. I have the same issue with a few sites, JavHD, MissAV and Naughtyblog off the top of my head. 

Error log for all 3 sites, https://paste.kodi.tv/cesupafuci

All 3 sites work with Windows. I also ran a clean install on a new Fire stick and had the same errors. I'd suspect it is an interaction with Kodi/Android causing the errors

**Comment by krtlsbrk** [comment]
> I think there is something going on with Kodi/Android. I have the same issue with a few sites, JavHD, MissAV and Naughtyblog off the top of my head.
> 
> Error log for all 3 sites, https://paste.kodi.tv/cesupafuci
> 
> All 3 sites work with Windows. I also ran a clean install on a new Fire stick and had the same errors. I'd suspect it is an interaction with Kodi/Android causing the errors

You're right. I just tested it in Windows and it's working without any error message.

**Comment by CeeBro082** [comment]
Is this a Real Debrid issue ? SeaPorn has stopped working for me and so has Seren, the movie and TV add-on

**Comment by krtlsbrk** [comment]
> Is this a Real Debrid issue ? SeaPorn has stopped working for me and so has Seren, the movie and TV add-on

Not a RD issue because it works on my PC.

**Comment by Liquidrag** [comment]
Also not working on a Mac.

**Comment by Gujal00** [comment]
It is due to the random user-agent being used by the addon triggering cloudflare block (not challenge) due to browser fingerprinting. The user-agent is generated once a week, so probably will work after a week when the UA changes. May need a different strategy for the UA generation

**Comment by krtlsbrk** [comment]
> It is due to the random user-agent being used by the addon triggering cloudflare block (not challenge) due to browser fingerprinting. The user-agent is generated once a week, so probably will work after a week when the UA changes. May need a different strategy for the UA generation

But how come it's not being triggered in Windows OS?

**Comment by jewbmx** [comment]
It works on windows because the ua code is based on windows. In order to fix the issue the dev should try coding it deeper like finding out the platform its used on then build a user agent that matches the platform.

**Comment by krtlsbrk** [comment]
> It works on windows because the ua code is based on windows. In order to fix the issue the dev should try coding it deeper like finding out the platform its used on then build a user agent that matches the platform.

Oh I see that makes sense. Hopefully dev will be able to fix this.

**Comment by jewbmx** [comment]
I did something similar last week in my addon by just saving my own 3 devices useragents and making a setting to pick which ones used in the addon lol, would of done it in the auto fashion but was too lazy to find the other possibilities but made it so other folks can share what might be missing.

**Comment by krtlsbrk** [comment]
> I did something similar last week in my addon by just saving my own 3 devices useragents and making a setting to pick which ones used in the addon lol, would of done it in the auto fashion but was too lazy to find the other possibilities but made it so other folks can share what might be missing.

How did you manage to change the user agent?

**Comment by TheMatrix31** [comment]
Any update on Naughty blog or Seaporn?

Both RD only and both haven't worked in a few weeks.

**Comment by 12asdfg12** [comment]
If you use RD, the videos will not work, as it seems that the RG links have been removed from the RD cache due to copyright infringement (and cannot be restored).
It may work if you use another debrid.

**Comment by krtlsbrk** [comment]
It has nothing to do with RD. As stated before it works on PC but doesn't work on Android.

**Comment by 12asdfg12** [comment]
What you are saying is a different issue, I don't use Kodi on Android so I can't verify this, but videos cannot be played with RD

![image](https://github.com/user-attachments/assets/048c2ca2-2e7c-468f-8042-3ca3edbac248)

**Comment by krtlsbrk** [comment]
Try with Kodi on Windows

**Comment by Slistbud** [comment]
Can someone please repair the ongoing issues with MissAv sink the beginning of November?  As mentioned in this thread, MissAv works perfectly fine in Kodi Cumination for Windows, however does not work on Android.  I’ve tried it on multiple Firesticks along with home network internet and cell phone networks via hotspots.  I’ve tried multiple VPN’s to no avail.  When attempting to play any video (on Android), I immediately receive an error saying “Uh Oh - It looks like the website is down”, then the pop-up log viewer shows a PythonToCppException error.  I’ve been hoping that someone could fix this issue that’s only present on Android.  Before anyone asks, I’m running the latest Kodi version on my Windows computer and Firestick.  In advance, THANKS for your help with resolving this issue.

**Comment by 12asdfg12** [comment]
Some sites (including MissAV or Naughtyblog) are blocked by Cloudflare when accessing videos.
It is curious that the main page works, and the blocking only occurs on Android, under Windows or Linux, there is no such problem.
At the moment I do not know of any solution for this issue.

**Comment by krtlsbrk** [comment]
Someone mentioned useragent workaround to bypass Cloudflare before
